src/components/Settings/Settings.tsx

```diff
@@ -235,49 +123,48 @@ export class Settings extends Component<Props, State> {
      * @param resourceType  ResourceTypes, the type of the control that changed
      */
     handleToggleChange = (isOn: boolean, resourceType: ResourceTypes) => {
-        let { cameraOn, audioOn } = this.state;
+        const { cameraOn, audioOn } = this.state;
+
+        let newCameraOn = resourceType === ResourceTypes.VIDEO ? isOn : cameraOn;
+        let newAudioOn = resourceType === ResourceTypes.AUDIO ? isOn : audioOn;
 
         // do not allow both camera and audio to be turned off
-        cameraOn = !isOn && resourceType === ResourceTypes.AUDIO ? true : cameraOn;
-        audioOn = !isOn && resourceType === ResourceTypes.VIDEO ? true : audioOn;
+        newCameraOn = !isOn && resourceType === ResourceTypes.AUDIO ? true : newCameraOn;
+        newAudioOn = !isOn && resourceType === ResourceTypes.VIDEO ? true : newAudioOn;
 
-        this.setState(
-            (prevState: State) => {
-                return {
-                    cameraOn: resourceType === ResourceTypes.VIDEO ? isOn : cameraOn,
-                    audioOn: resourceType === ResourceTypes.AUDIO ? isOn : audioOn,
-                    screenOn:
-                        resourceType === ResourceTypes.SCREEN_SHARE ? isOn : prevState.screenOn
-                };
-            },
-            () => {
-                this.saveSettings();
-            }
-        );
-    };
+        this.setState((prevState: State) => {
+            return {
+                cameraOn: newCameraOn,
+                audioOn: newAudioOn,
+                screenOn: resourceType === ResourceTypes.SCREEN_SHARE ? isOn : prevState.screenOn
+            };
+        });
```

Comment:
please try to avoid those long conditions. it is really hard to understand them.
I suggest doing something like:
```
let { cameraOn } = this.state;

if(resourceType === ResourceTypes.VIDEO){
  cameraOn = isOn; // you can change this blocked scope variable, it is local
}
if(!isON && resourceType === ResourceTypes.AUDIO){
  cameraOn = true;
}
this.setState((prevState) => {
  return {
    ...prevState,
    cameraOn
  };
});
```

Reply:
it's needed.. the logic here need to consider the state, and new values. what you wrote is not equivalent to the code